nesta/core/orms/nih_orm.py

```diff
@@ -2,116 +2,122 @@
-    project_end = Column(VARCHAR(10))
-    project_terms = Column(TEXT)
-    project_title = Column(VARCHAR(200))
+    pi_ids = Column(JSON)
+    pi_names = Column(JSON)
+    program_officer_name = Column(TEXT)
+    project_start = Column(DATETIME)
+    project_end = Column(DATETIME)
+    project_terms = Column(JSON)
+    project_title = Column(TEXT)
     serial_number = Column(VARCHAR(6))
     study_section = Column(VARCHAR(4))
-    study_section_name = Column(VARCHAR(95))
+    study_section_name = Column(TEXT)
     suffix = Column(VARCHAR(6))
     support_year = Column(VARCHAR(2))
     direct_cost_amt = Column(INTEGER)
     indirect_cost_amt = Column(INTEGER)
     total_cost = Column(INTEGER)
     subproject_id = Column(INTEGER)
     total_cost_sub_project = Column(INTEGER)
-    nih_spending_cats = Column(VARCHAR(2232))
+    nih_spending_cats = Column(JSON)
 
 
 class Abstracts(Base):
     __tablename__ = 'nih_abstracts'
 
-    application_id = Column(INTEGER, primary_key=True)
+    application_id = Column(INTEGER, primary_key=True, autoincrement=False)
     abstract_text = Column(TEXT)
 
+
 class Publications(Base):
     __tablename__ = 'nih_publications'
 
-    pmid = Column(INTEGER, primary_key=True)
-    author_name = Column(VARCHAR(166))
+    pmid = Column(INTEGER, primary_key=True, autoincrement=False)
+    author_name = Column(TEXT)
     affiliation = Column(TEXT)
-    author_list = Column(TEXT)
-    country = Column(VARCHAR(25))
+    author_list = Column(JSON)
+    country = Column(VARCHAR(50), index=True)
     issn = Column(VARCHAR(9))
     journal_issue = Column(VARCHAR(75))
-    journal_title = Column(VARCHAR(282))
-    journal_title_abbr = Column(VARCHAR(108))
+    journal_title = Column(VARCHAR(400), index=True)
+    journal_title_abbr = Column(VARCHAR(200))
     journal_volume = Column(VARCHAR(100))
     lang = Column(VARCHAR(3))
-    page_number = Column(VARCHAR(138))
-    pub_date = Column(VARCHAR(23))
-    pub_title = Column(TEXT)
-    pub_year = Column(INTEGER)
-    pmc_id = Column(INTEGER)
+    page_number = Column(VARCHAR(200))
+    pub_date = Column(DATETIME)
+    pub_title = Column(VARCHAR(400), index=True)
+    pub_year = Column(INTEGER, index=True)
+    pmc_id = Column(INTEGER, index=True)
+
      
 class Patents(Base):
     __tablename__ = 'nih_patents'
     
-    patent_id = Column(VARCHAR(9), primary_key=True)
-    patent_title = Column(VARCHAR(284))
-    project_id = Column(VARCHAR(11), index=True)
-    patent_org_name = Column(VARCHAR(50))
+    patent_id = Column(VARCHAR(20), primary_key=True)
+    patent_title = Column(TEXT)
+    project_id = Column(VARCHAR(50), index=True)
+    patent_org_name = Column(TEXT)
+
 
-# class LinkTables(Base):
-#     __tablename__ = 'nih_linktables'
+class LinkTables(Base):
+    __tablename__ = 'nih_linktables'
 
-#     pmid = Column(INTEGER, primary_key=True)
-#     project_number = Column(VARCHAR(11), index=True)
+    pmid = Column(INTEGER, primary_key=True, autoincrement=False)
```

Comment:
so the field names and table names are automatically generated from the data collected from the NiH data dump, and so e.g. `pmid` is unfortunately their choice. Agreed that we need a data-dictionary per dataset. This is in scope for DAPS2, with the idea that we can introduce a new "`Column`" type which will also have a field called `description`, which will automatically end up in a separate `data_dictionary` table with fields like `table_name` `field_name` and `description`.

Reply:
this will be in addition to the open ontology (e.g. schema.org), although it's to be discussed whether that's in scope for May 2021, or further down the line